### ISSUE TITLE
Issues Annotating Transcripts w/ Gaps or X Codons

### ISSUE THREAD

**Issue by Akeeeeshi** [biocommons/hgvs#428]
Gene: NBPF1
Transcript: NM_017940.4
Example g. to Test: NC_000001.10:g.16890441C>T

Issue: Everything from the gc_offset position and onwards (everything 3' of position)  in transcript fails to annotate with HGVS information. 
gc_offset position for this transcript = chr1: 16893674

This corresponds exactly to an unresolved region from the source RefSeq GBS files. See picture below for the that exact exon start and end boundary falling just after the gc_offset. 

<img width="600" alt="screen shot 2017-04-19 at 5 27 07 pm" src="https://cloud.githubusercontent.com/assets/28007814/25388915/1a83f27e-299d-11e7-91fe-43c9f87201ee.png">

I was just wondering if this is something that can possibly fixed (unlikely since this is a problem with the reference sequence) or if this can be flagged in any sort of way in order to prevent throwing an error. 

Other genes I've seen this issue in SHANK2, CDK11B, NBPF9 (there are likely more)

**Comment by reece** [comment]
The general problem here is really with UTA, the source of alignments used by hgvs.

The problem is that UTA assumes that transcripts are aligned exon-wise. When an exon spans an assembly gap or contains intervening sequence, the exon corresponds to two or more aligned regions. UTA doesn't accommodate that. hgvs attempts to recognize such cases and raise an exception, but it's possible we don't cover all of these cases.

Fixing the underlying issue is complex. See #385, biocommons/uta#198, and biocommons/uta#200.

It would be extremely helpful if you would add specific examples to this issue in a form roughly like this:

```
(venv) snafu$ hgvs-shell 
In [1]: %doctest_mode
>>> hgvs.__version__
'1.0.0.post4.dev1+h7c049c8b37e4.d20170414'
>>> str(hdp)
'UTA_postgresql <data_version:uta_20170117; schema_version:1.1; application_name=hgvs-shell; url=postgresql://anonymous:anonymous@uta.biocommons.org/uta/uta_20170117; sequences-from=seqfetcher>'
>>> v = hp.parse_hgvs_variant("NC_000001.10:g.16890441C>T")
>>> am37.g_to_c(v, "NM_017940.4")
...
HGVSInvalidIntervalError: start or end or both are beyond the bounds of transcript record
```

(`hgvs-shell` is installed with hgvs. It initializes a UTA dataprovider (hdp), Parser (hp), Validator (hv), and 37 and 38 AssemblyMappers (am37 and am38). This makes it easy for me to reproduce errors and incorporate them into our regression test.)

**Comment by Akeeeeshi** [comment]
Some examples 

**SHANK2**

`In [4]: v3 = hp.parse_hgvs_variant("NC_000011.9:g.70407467G>A")`
`In [5]: variantmapper.g_to_c(v3, "NM_012309.3")`
`...`
`HGVSInvalidIntervalError: start or end or both are beyond the bounds of transcript record`


**MUC2**

`In [4]:v4 = hp.parse_hgvs_variant("NC_000011.9:g.1092890C>T")`
`In [5] variantmapper.g_to_c(v4, "NM_002457.2")`
`...`
`HGVSInvalidIntervalError: start or end or both are beyond the bounds of transcript record`


**NBPF9**

`In [8]: v4 = hp.parse_hgvs_variant("NC_000011.9:g.1092890C>T")`
`In [9]: variantmapper.g_to_c(v4, "NM_002457.2")`
`...`
`HGVSDataNotAvailableError: No tx_info for (tx_ac=NM_001277444,alt_ac=NC_000001.10,alt_aln_method=splign)`

**Comment by reece** [comment]
Thanks for the examples. There's something funny with the last one: the exception displays a different transcript than what you specified. I hope that's not really happening! When I try it, I get the same HGVSInvalidIntervalError. My first guess is that you're using an older hgvs that by default uses an older UTA that's missing that transcript.

**Comment by Akeeeeshi** [comment]
Sorry should have specified I am using UTA20161216 and hgvs 0.4

**Comment by reece** [comment]
Ah, okay. Can you upgrade to 1.0.0? I don't plan any fixes to the 0.4 series.

Regarding your original comment:
> I was just wondering if this is something that can possibly fixed (unlikely since this is a problem with the reference sequence) or if this can be flagged in any sort of way in order to prevent throwing an error.

My only suggestion (for 0.4 or 1.0) is to wrap operations in try...except blocks, where you catch HGVSDataNotAvailableError or HGVSInvalidIntervalError specifically, or just catch HGVSError generally.

**Comment by reece** [comment]
Closing this issue as a duplicate of #385. Discussion should continue in biocommons/uta#198.
